src/EventHandler/Message.php

```diff
@@ -137,9 +142,122 @@ public function __construct(
+    {
+        $result = $this->getClient()->methodCallAsyncRead(
+            'messages.updatePinnedMessage',
+            [
+                'peer' => $this->chatId,
+                'id' => $this->id,
+                'pm_oneside' => $pmOneside,
+                'silent' => $silent,
+                'unpin' => true
+            ]
+        );
+        return $this->getClient()->wrapUpdate($result);
+    }
+
+    /**
+     * Get our reaction on message return null if message deleted.
+     *
+     * @return list<string|int>|null
+     */
+    public function getReactions(): ?array
+    {
+        if (!$this->reactionsCached) {
+            $this->reactionsCached = true;
+            $me = $this->getClient()->getSelf()['id'];
+            $myReactions = \array_filter(
+                $this->getClient()->methodCallAsyncRead(
+                    'messages.getMessageReactionsList',
+                    [
+                        'peer' => $this->chatId,
+                        'id' => $this->id
+                    ]
+                )['reactions'],
+                fn (array $r): bool => \intval($r['peer_id']['user_id'] ?? $r['peer_id']['channel_id']) == $me
+            );
+            $this->reactions = \array_map(fn (array $r) => $r['reaction']['emoticon'] ?? $r['reaction']['document_id'], $myReactions);
+        }
+        return $this->reactions;
+    }
+
+    /**
+     * Add reaction to message.
+     *
+     * @param list<string|int> $reaction Array of Reaction
+     * @param bool $big Whether a bigger and longer reaction should be shown
+     * @param bool $addToRecent Add this reaction to the recent reactions list.
+     */
+    public function addReaction(array $reaction, bool $big = false, bool $addToRecent = true): ?Update
+    {
+        $result = $this->getClient()->methodCallAsyncRead(
+            'messages.sendReaction',
+            [
+                'peer' => $this->chatId,
+                'msg_id' => $this->id,
+                'reaction' => \array_map(fn ($reactions) => \is_int($reactions) ? ['_' => 'reactionCustomEmoji', 'document_id' => $reactions] : ['_' => 'reactionEmoji', 'emoticon' => $reactions], $reaction),
+                'big' => $big,
+                'add_to_recent' => $addToRecent
+            ]
+        );
+        $this->reactions += $reaction;
+        return $this->getClient()->wrapUpdate($result);
+    }
+
+    /**
+     * Delete reaction from message.
+     *
+     * @param string|int $reaction string or int Reaction
+     */
+    public function delReaction(int|string $reaction): ?Update
+    {
+        $this->getReactions();
+        unset($this->reactions[\array_search($reaction, $this->reactions)]);
+        $r = \array_map(fn ($reactions) => \is_int($reactions) ? ['_' => 'reactionCustomEmoji', 'document_id' => $reactions] : ['_' => 'reactionEmoji', 'emoticon' => $reactions], $this->reactions);
+        $r[]= ['_' => 'reactionEmpty'];
```

Comment:
I don't think this is needed?

Reply:
You know if we want to remove our reaction in past we just need to re reacted with same reaction too remove it but now we need to use empty reaction + old reaction(for premium users) to remove reaction i test it last night

Reply:
oki!